generatehtml.py

```diff
@@ -6,15 +6,32 @@
 import codecoverage
 
 
-def generate_html(data_folder):
+def generate_html(data_folder, ignore_third_party=True):
     with open('{}/diff.json'.format(data_folder), 'r') as report:
         parsed_json = json.load(report)
 
     file_obj = open('{}/output.info'.format(data_folder), 'w')
 
     source_files = parsed_json['source_files']
     file_obj.write('TN\n')
+
+    third_party_paths = []
+    with open('mozilla-central/tools/rewriting/ThirdPartyPaths.txt') as f:
+        third_party_paths = list(f)
+
     for source_file in source_files:
+        if ignore_third_party is True:
+            ignore = False
```

Comment:
You could rewrite this in a nicer way:
```Python
# If we want to ignore third-party files and the source file is third-party, skip it.
if ignore_third_party and any(source_file['name'].startswith(path.strip('\n')) for path in third_party_paths):
    continue
```

And I would actually move the stripping part to where you read the file:
```Python
    with open('mozilla-central/tools/rewriting/ThirdPartyPaths.txt') as f:
        third_party_paths = [path.strip('\n') for path in f]
```

Maybe we should actually move this in crawler.py, before computing the diff, so we don't spent time computing the diff for paths that then we ignore.

Reply:
Probably, we can move it here: https://github.com/marco-c/coverage-crawler/blob/c401e8b7d15e117468f143ac1336d2c5acefff09/diff.py#L23.
Thanks for suggesting a nice way of writing the condition.

Reply:
I'd move it in crawler.py, before calculating the diff.
ignore_third_party is unrelated to diffing, so I wouldn't make it an option of diff.

Reply:
But I don't work with source file names in crawler: only in `diff.py` and `generatehtml.py`.

Reply:
You could add it though, at https://github.com/marco-c/coverage-crawler/blob/master/crawler.py#L261 you have the report objects (`baseline_report` and `report`), so you can filter them.
You could also create a new module, `filter`, which for now will only contain the ignore_third_party filter.